inc/saitunnel.h

```diff
@@ -0,0 +1,444 @@
+
+    /** tunnel dscp mode (pipe or uniform model) [sai_tunnel_dscp_mode_t]
+    *  (MANDATORY_ON_CREATE when SAI_TUNNEL_ATTR_TYPE=SAI_TUNNEL_IPINIP,SAI_TUNNEL_IPINIP_GRE,
+    *  SAI_TUNNEL_IPINIP_GRE_AND_KEY) (CREATE_ONLY) */
+    SAI_TUNNEL_ATTR_DECAP_DSCP_MODE,
+
+    /** Custom range base value */
+    SAI_TUNNEL_ATTR_CUSTOM_RANGE_BASE = 0x10000000
+
+} sai_tunnel_attr_t;
+
+/**
+ * Routine Description:
+ *    @brief Create tunnel 
+ *
+ * Arguments:
+ *    @param[out] tunnel_id - tunnel id
+ *    @param[in] attr_count - number of attributes
+ *    @param[in] attr_list - array of attributes
+ *
+ * Return Values:
+ *    @return SAI_STATUS_SUCCESS on success
+ *            Failure status code on error
+ *
+ */
+typedef sai_status_t (*sai_create_tunnel_fn)(
+    _Out_ sai_object_id_t* tunnel_id,
+    _In_ uint32_t attr_count,
+    _In_ const sai_attribute_t *attr_list
+    );
+
+/**
+ * Routine Description:
+ *    @brief Remove tunnel 
+ *
+ * Arguments:
+ *    @param[in] tunnel_id � tunnel id 
+ *
+ * Return Values:
+ *    @return SAI_STATUS_SUCCESS on success
+ *            Failure status code on error
+ */
+typedef sai_status_t (*sai_remove_tunnel_fn)(
+    _In_ sai_object_id_t tunnel_id
+    );
+
+/**
+ * Routine Description:
+ *    @brief Set tunnel attribute
+ *
+ * Arguments:
+ *    @param[in] tunnel_id - tunnel id
+ *    @param[in] attr - attribute
+ *
+ * Return Values:
+ *    @return SAI_STATUS_SUCCESS on success
+ *            Failure status code on error
+ */
+typedef sai_status_t (*sai_set_tunnel_attribute_fn)(
+    _In_ sai_object_id_t tunnel_id,
+    _In_ const sai_attribute_t *attr
+    );
+
+
+/**
+ * Routine Description:
+ *    @brief Get tunnel attributes
+ *
+ * Arguments:
+ *    @param[in] tunnel _id - tunnel id
+ *    @param[in] attr_count - number of attributes
+ *    @param[inout] attr_list - array of attributes
+ *
+ * Return Values:
+ *    @return SAI_STATUS_SUCCESS on success
+ *            Failure status code on error
+ */
+typedef sai_status_t (*sai_get_tunnel_attribute_fn)(
+    _In_ sai_object_id_t tunnel_id,
+    _In_ uint32_t attr_count,
+    _Inout_ sai_attribute_t *attr_list
+    );
+
+typedef enum _sai_tunnel_table_entry_type_t
+{
+    /** tunnel table point to point entry match on dst & src ip & tunnel type  **/
+    SAI_TUNNEL_TABLE_ENTRY_P2P,
+
+    /** tunnel table point to multi point entry match on dst ip  & tunnel type  **/
+    SAI_TUNNEL_TABLE_ENTRY_P2MP,
+
+} sai_tunnel_table_entry_type_t;
+
+typedef enum _sai_tunnel_table_entry_attr_t
+{
+    /** READ-ONLY */
+
+    /** READ-WRITE */
+
+    /** tunnel virtual router id [sai_object_id_t] (MANDATORY_ON_CREATE|CREATE_ONLY)*/
+    SAI_TUNNEL_TABLE_ENTRY_ATTR_VR_ID,
+
+    /** tunnel entry type [sai_tunnel_table_entry_type_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
+    SAI_TUNNEL_TABLE_ENTRY_ATTR_TYPE,
+
+    /** tunnel termination ip address [sai_ip_address_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
+    SAI_TUNNEL_TABLE_ENTRY_ATTR_DST_IP,
```

Comment:
is this the same as SAI_TUNNEL_ATTR_SRC_IP? If yes, can we remove this one here?